open-company/open-company-web
# Fix CLJS error reporting in prod contexts

**This is based on #86, only commit that's not part of #86 is https://github.com/open-company/open-company-web/pull/87/commits/12b644b95dc706ef37e4a62badb31fd891da493c**

This PR fixes issues with tracking exceptions on staging/beta (and other production-like contexts). Previously Raven was not setup correctly due to missing externs.

I now switched to using cljsjs/raven which comes with externs.

To Test:

This should return `true`:
```js
Raven.isSetup()
```

This should trigger an exception being logged in [sentry](https://app.getsentry.com/opencompany/oc-staging-web/)
```js
try {
      throw "an actual error from staging!"; 
    } catch(e) {
      Raven.captureException(e)
}
```

Exceptions triggered with just `throw` from the console are not processed by Sentry. I tested with the following code added to our code that exceptions during runtime are caught and sent to Sentry:
```clj

(.setTimeout js/window
             (fn []
               (js/console.info "Throwing Exception...")
               (throw "an error thrown by our application code"))
             3000)
```
